src/kOS.Safe/Communication/Connection.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+
+namespace kOS.Safe.Communication
+{
+    [kOS.Safe.Utilities.KOSNomenclature("Connection")]
+    public abstract class Connection<S> : Structure where S : SharedObjects
```

Comment:
Okay that makes a lot more sense then.  So then you only have to store the `shared` member in the derived class because none of the parent stuff that's defined in the abstract base class explicitly tries using it - shared is only used in the bodies of some of the overriding methods and properties in the derived class, not in the base class.